ios/Classes/Module/FacebookModule.m

```diff
@@ -162,11 +162,21 @@ - (void)logPurchase:(NSArray<id> *_Nonnull)purchase
 {
   ENSURE_TYPE([purchase objectAtIndex:0], NSNumber);
   ENSURE_TYPE([purchase objectAtIndex:1], NSString);
+  ENSURE_TYPE([purchase objectAtIndex:2], NSDictionary);
 
   NSNumber *amount = [purchase objectAtIndex:0];
   NSString *currency = [TiUtils stringValue:[purchase objectAtIndex:1]];
+  NSDictionary *parameters = [purchase objectAtIndex:2];
```

Comment:
If there is only 2 parameters, it will crash here. Please add proper check on parameter count.